### PR TITLE
gui: Modal tweaks

### PR DESCRIPTION
**Pull request by norgeous**
## Remove show / hide logic for:
- Settings > Anonymous Usage Reporting > (Preview)
- Edit Folder > Ignore Patterns
- Edit Device > Show QR
- About button
- upgradeMajor() in index.html

## Remove crazy amount of view directives and combine into one!
- Previously we had 17 files that were clones of:
```js
angular.module('syncthing.core')
    .directive('httpErrorDialog', function () {
        return {
            restrict: 'A',
            templateUrl: 'syncthing/core/httpErrorDialogView.html'
        };
});
```

- Now all views covered by ng-include! Template files can now be loaded with:

```html
<ng-include src="'syncthing/core/networkErrorDialogView.html'"></ng-include>
```

- I have removed all the modules, as they were just for loading templates and contained no logic

## Redid existing modals to be compatible with new modalDirective.

- Removed footer from modal.html (containing a close button), 
- each modal's footer is now in each template html file.
- every modal uses modalDirective, with `<modal></modal>`

## Testing
open the web dev console and type:
```js
$('#ur').modal()
$('#failed').modal()
$('#needed').modal()
$('#majorUpgrade').modal()
```

### CODE REVIEW

**Comment by calmh** [comment]
Neat

File: gui/default/syncthing/settings/settingsModalView.html [hunk 1]
```diff
@@ -119,7 +119,7 @@ <h4 class="modal-title">
             <div class="form-group">
               <div class="checkbox">
                 <label>
-                  <input id="UREnabled" type="checkbox" ng-model="tmpOptions.urEnabled"> <span translate>Anonymous Usage Reporting</span> (<a translate ng-click="showURPreview()" href="#">Preview</a>)
+                  <input id="UREnabled" type="checkbox" ng-model="tmpOptions.urEnabled"> <span translate>Anonymous Usage Reporting</span> (<a href="" translate data-toggle="modal" data-target="#urPreview">Preview</a>)
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/settings/settingsModalView.html]: Is the empty href actually needed?

File: gui/default/syncthing/core/syncthingController.js [hunk 2]
```diff
@@ -1460,29 +1454,14 @@ angular.module('syncthing.core')
...
-                                });
-                        });
+                    var textArea = $('#editIgnores textarea');
+                    textArea.val(data.ignore.join('\n'));
+                    $('#editIgnores').modal()
+                    textArea.focus();
                 })
-                .then(function () {
-                    $('#editIgnoresButton').removeAttr('disabled');
-                });
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: Why removing this?

File: gui/default/syncthing/core/syncthingController.js [hunk 3]
```diff
@@ -1460,29 +1454,14 @@ angular.module('syncthing.core')
...
-                                    $('#editFolder').modal();
-                                })
-                                .one('shown.bs.modal', function () {
-                                    textArea.focus();
-                                });
-                        });
+                    var textArea = $('#editIgnores textarea');
+                    textArea.val(data.ignore.join('\n'));
+                    $('#editIgnores').modal()
+                    textArea.focus();
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: Still needs shown as focus on a not yet visible element does not work

File: gui/default/index.html [hunk 4]
```diff
@@ -628,23 +628,23 @@ <h4 class="panel-title">
...
-  <div edit-folder-modal></div>
-  <div edit-ignores-modal></div>
-  <div settings-modal></div>
-  <div advanced-settings-modal></div>
-  <div usage-report-modal></div>
-  <div usage-report-preview-modal></div>
-  <div needed-files-modal></div>
-  <div failed-files-modal></div>
-  <div about-modal></div>
+  <view idxxx="network-error-dialog"       template-url="syncthing/core/networkErrorDialogView.html"></view>
```
Comment (AudriusButkevicius) [inline gui/default/index.html]: Why xxx?

Comment (calmh) [inline gui/default/index.html]: I think this is the large "work in progress" thing :)

File: gui/default/syncthing/settings/settingsModalView.html (hunk 1 above)
Comment (calmh) [inline gui/default/syncthing/settings/settingsModalView.html]: IIRC this is needed to get the correct mouse pointer on links.

Comment (norgeous) [inline gui/default/syncthing/settings/settingsModalView.html]: yeah, i saw it used elsewhere (Actions menu, i think?)

File: gui/default/index.html (hunk 4 above)
Comment (norgeous) [inline gui/default/index.html]: xxx pron ftw. jk. yeah just set that in a moment of, ahem, haste...

File: gui/default/syncthing/core/syncthingController.js (hunk 2 above)
Comment (norgeous) [inline gui/default/syncthing/core/syncthingController.js]: thought it wasn't needed at the time. idk why. shall add back

File: gui/default/syncthing/core/syncthingController.js (hunk 3 above)
Comment (norgeous) [inline gui/default/syncthing/core/syncthingController.js]: you are correct sir

**Comment by norgeous** [comment]
OK? I think this is good now... Sorry there is so much to review, but I think it will be worth it for the readability and consolidation this provides. I have done some testing and it looks like it works, should be careful though I guess, as a lot was edited.

**Comment by norgeous** [comment]
converting this:
```html
<div id="editDevice" class="modal fade" tabindex="-1">
  <div class="modal-dialog modal-lg">
    <div class="modal-content">
      <div class="modal-header">
        <h4 class="modal-title">
          <span ng-show="!editingExisting">
            <div class="panel-icon">
              <span class="fa fa-desktop"></span>
            </div>
            <span translate>Add Device</span>
          </span>
          <span ng-show="editingExisting">
            <div class="panel-icon">
              <span class="fa fa-pencil"></span>
            </div>
            <span translate>Edit Device</span>
          </span>
        </h4>
      </div>
```
to this:
```html
<modal id="editDevice" status="default" icon="{{editingExisting ? 'pencil' : 'desktop'}}" heading="{{editingExisting ? 'Edit Device' : 'Add Device' | translate}}" large="yes" closeable="yes">
```
was a big win IMO!

**Comment by AudriusButkevicius** [comment]
@st-review lgtm

**Comment by st-review** [comment]
@AudriusButkevicius: Noted! Need another LGTM or explicit merge command.

**Comment by calmh** [comment]
I want to just take this for a spin and do some testing, but might not have time to do so today. It's not forgotten anyway :)

**Comment by calmh** [comment]
@st-review lgtm

**Comment by calmh** [comment]
Thanks 👍

**Comment by st-review** [comment]
:ok_hand: Merged as de298da53240a6023a0df3d0fe45c829c3792cf5. Thanks, @norgeous!

**Pull request closed without merging**